tripal/src/Services/TripalPublish.php

```diff
@@ -485,6 +493,59 @@ protected function addFixedTypeValues(&$search_values) {
     }
   }
 
+  /**
+   * Adds to the search values array any remaining property values.
+   *
+   * @param array $seach_values
+   */
+  protected function addNonRequiredValues(&$search_values) {
```

Comment:
This method feels much more complicated then it needs to be :thinking: but it took me a couple of reads to see why. I think we just need to add a helper method that checks if a field is supported :-) That helper can then be used both here and in the addRequiredValues() method which reduces duplicated code and makes the logic in both these methods simpler.

With this change we check if the field is supported before looping through the properties and we don't need any checks for supportedness in the loop. We can also move defining $field_definition and $field_class outside the property loop.

Something like this:
```php
protected function checkFieldIsSupported(string $field_name) {

  // This property may be part of a field which has already been marked
  // as unsupported. If so then it won't be in the field_info and we
  // should skip it.
  if (!array_key_exists($field_name, $this->field_info)) {
    // Add it to the list of unsupported fields just in case
    // it wasn't added before...
    $this->unsupported_fields[$field_name] = $field_name;
    return FALSE;
  }

  return TRUE;
}
```

Reply:
Updated with commit https://github.com/tripal/tripal/pull/1952/commits/0eea7d250468a4350a29a1297b1bf4b6f121ed76